src/gather.jl

```diff
@@ -63,3 +63,17 @@ function NNlib.gather!(dst::AnyCuArray, src::AnyCuArray, idx::AnyCuArray)
     kernel(args...; threads=threads, blocks=blocks)
     return dst
 end
+
+function NNlib.gather(src::AnyCuArray{Tsrc, Nsrc}, 
+                      idx::AnyCuArray{Tidx, Nidx}) where 
+            {Tsrc, Nsrc, Nidx, Tidx}
+    M = NNlib.typelength(Tidx) 
+    dstsize = (size(src)[1:Nsrc-M]..., size(idx)...)
+    dst = similar(src, Tsrc, dstsize)
+    return NNlib.gather!(dst, src, idx)
+end
```

Comment:
Why we need this method? The one in Nlib is not enough?

Reply:
I take it from NNlib. The dispatching rule needs to distinguish `(::AnyCuArray, ::AnyCuArray)` from `(::AnyCuArray, ::AbstractArray)`.